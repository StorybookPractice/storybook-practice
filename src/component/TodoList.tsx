import TodoItem, { TodoItemProps } from "./TodoItem";
import "../stories/todo-item.css";

interface TodoListProp {
  todoItems: TodoItemProps[];
}

export default function TodoList({ todoItems }: TodoListProp) {
  return (
    <>
      {todoItems.map((todoItem: TodoItemProps) => (
        <div key={`${todoItem.todoText}-wrap`} className="storybook-todo-list--container">
          <TodoItem key={`${todoItem.todoText}-component`} todoText={todoItem.todoText} isDone={todoItem.isDone} readOnly={todoItem.readOnly} />
        </div>
      ))}
    </>
  );
}
