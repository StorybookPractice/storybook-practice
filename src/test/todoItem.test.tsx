import TodoList from "../component/TodoItem";
import { it, describe, expect } from "vitest";
describe("TodoItem 테스트", () => {
  it("TodoItem 스냅샷 테스트", () => {
    const todoItem = TodoList({ todoText: "test", isDone: false, readOnly: false });
    expect(todoItem).toMatchSnapshot();
  });
});
