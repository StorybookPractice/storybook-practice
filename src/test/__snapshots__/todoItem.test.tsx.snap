// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TodoItem 테스트 > TodoItem 스냅샷 테스트 1`] = `
<div
  className="storybook-todo-list--container"
  style={
    {
      "backgroundColor": "#ffffff",
      "border": "1px solid #FF8F50",
    }
  }
>
  <div
    className="storybook-todo-list--wrap-content"
  >
    <div
      className="storybook-todo-list--icon"
    >
      <Icon
        fill="#FF8F50"
        size={16}
        type="CIRCLE"
      />
    </div>
    <input
      className="storybook-todo-list--content"
      placeholder="할 일을 작성해 주세요."
      readOnly={false}
      style={
        {
          "backgroundColor": "#ffffff",
          "color": "#000000",
        }
      }
      type="text"
      value="test"
    />
  </div>
  <Icon
    fill="#FF8F50"
    size={16}
    type="TRASH"
  />
</div>
`;
